src/main/scala/org/ergoplatform/http/api/TransactionsApiRoute.scala

```diff
@@ -87,4 +124,120 @@ case class TransactionsApiRoute(readersHolder: ActorRef,
     ApiResponse(getMemPool.map(_.getExpectedWaitTime(fee,txSize).asJson))
   }
 
+  /** Unconfirmed Txs */
+
+  /** Check whether given transaction is present in mempool without returning it */
+  def unconfirmedContainsR: Route = (pathPrefix("unconfirmed") & head & modifierId) { modifierId =>
+    ApiResponse(getMemPool.map(_.modifierById(modifierId)))
+  }
+
+  /** Get unconfirmed transactions at given offset and limit*/
+  def getUnconfirmedTransactionsR: Route = (path("unconfirmed") & get & txPaging) { (offset, limit) =>
+    ApiResponse(getUnconfirmedTransactions(offset, limit))
+  }
+
+  /** Get unconfirmed transaction by its id */
+  def getUnconfirmedTxByIdR: Route =
+    (pathPrefix("unconfirmed" / "byTransactionId") & get & modifierId) { modifierId =>
+      ApiResponse(getMemPool.map(_.modifierById(modifierId)))
+    }
+
+  /** Collect all transactions which inputs or outputs contain given ErgoTree hex */
+  def getUnconfirmedTxByErgoTreeR: Route =
+    (pathPrefix("unconfirmed" / "byErgoTree") & post & entity(as[Json])) { body =>
```

Comment:
I think paging should be implemented for this query, for some trees there could be thousands of transactions